src/jimp2/wireworld/z8/datamangment/FactoryOfCustomElements.java

```diff
@@ -42,67 +37,74 @@ private void initializeElements() throws IOException {
+                    continue;
                 }
 
                 //reading input point of custom element
                 line = reader.readLine();
-                w = line.split("\\s+");
-                if (w.length == 2)
-                    inConnectorPoint = new Point(Integer.parseInt(w[0]), Integer.parseInt(w[1]));
-                else
+                numbers = line.split("\\s+");
+                if (numbers.length == 2)
+                    inConnectorPoint = new Point(Integer.parseInt(numbers[0]), Integer.parseInt(numbers[1]));
+                else {
                     System.err.println("Wrong data in custom element file (input point)");
-
-                CustomElement cs = new CustomElement(width,height,files[i],inConnectorPoint);
+                    continue;
+                }
+                CustomElement customElement = new CustomElement(width,height,files[i],inConnectorPoint);
 
                 //reading table
                 int row = 0;
                 while((line = reader.readLine()) != null) {
-                    w = line.split("\\s+");
-                    if (w.length != width)
+                    numbers = line.split("\\s+");
+                    if (numbers.length != width) {
                         System.err.println("There is not enough points (width)");
+                        continue;
+                    }
                     else {
                         for (int column = 0; column < width; column++) {
-                            if (Integer.parseInt(w[column]) == 0)
-                                cs.cells[column][row].setState(State.EMPTY);
-                            if (Integer.parseInt(w[column]) == 1)
-                                cs.cells[column][row].setState(State.CONDUCTOR);
-                            if (Integer.parseInt(w[column]) == 2)
-                                cs.cells[column][row].setState(State.HEAD);
-                            if (Integer.parseInt(w[column]) == 3)
-                                cs.cells[column][row].setState(State.TAIL);
+                            int number =  Integer.parseInt(numbers[column]);
+                            if (number == 0)
+                                customElement.cells[column][row].setState(State.EMPTY);
+                            if (number == 1)
+                                customElement.cells[column][row].setState(State.CONDUCTOR);
+                            if (number == 2)
+                                customElement.cells[column][row].setState(State.HEAD);
+                            if (number == 3)
+                                customElement.cells[column][row].setState(State.TAIL);
                         }
                     }
                     row++;
                 }
-                availableCustomElements.put(files[i].substring(0,files[i].length()-4),cs);
+
+                availableCustomElements.put(files[i].substring(0,files[i].length()-4),customElement);
```

Comment:
to na pewno jest nazwa bez ścieżki Data/CustomElements? pytam na wszelki wypadek

Reply:
![image](https://user-images.githubusercontent.com/76851989/121331740-750ab980-c917-11eb-9cf0-214d6a893ae5.png)
Z ekranu do debugowania

Reply:
o to super, sorka